store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReader.java

```diff
@@ -90,6 +91,33 @@ public T readNextRow() throws IOException, InterruptedException {
     return currentReader.getCurrentValue();
   }
 
+  /**
+   * Read and return next string row object
+   */
+  public Object[] readNextStringRow() throws IOException, InterruptedException {
```

Comment:
I don't know in C layer , how to read int, double, date from mix object[] that includes string, int, date. only can read data string now

Reply:
I convert double and date to string first, then return string[]. So the values didn't change.

Reply:
readNextStringRow is not a standard API I think. It can not work with nested complex data types. Like Array[Array[...]], Struct[Array]. 

I suggest to add a better API like CarbonRow wrapping on object[] and give reader utility methods like getInt, getString in CarbonRow .

If it can't be finished in this PR scope, temporarily you can move this string making logic to JNI layer code so that it can be improved later. **Adding in java SDK layer will add more confusion to users**

Reply:
I am developing in local and will raise a new PR for it. CSDK only need read Array[String] now.

Reply:
OK.. Please add this limitation as only single dimension Array is supported in method signature and remove after enhancing with CarbonRow based API in other PR

Reply:
OK，I added and will raise new PR for enhancing！